cmd/kubeadm/app/cmd/upgrade/common.go

```diff
@@ -72,6 +73,15 @@ func loadConfig(cfgPath string, client clientset.Interface, skipComponentConfigs
 	// This is probably 90% of the time. So we handle it first.
 	if cfgPath == "" {
 		cfg, err := configutil.FetchInitConfigurationFromCluster(client, printer, logPrefix, false, skipComponentConfigs)
+		// In case we fetch a configuration from the cluster, mutate the ImageRepository field
+		// to be 'registry.k8s.io', if it was 'k8s.gcr.io'.
+		// TODO: Remove this in 1.26
+		// https://github.com/kubernetes/kubeadm/issues/2671
+		if err == nil {
+			if err := uploadconfig.MutateImageRepository(cfg, client); err != nil {
+				return nil, false, err
+			}
+		}
```

Comment:
Do we need to make similar changes at these locations as well?
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/cmd/upgrade/diff.go#L121
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/cmd/upgrade/node.go#L142

Reply:
/hold

Feel free to cancel hold @neolit123

Reply:
good point @SataQiu 

so for "diff" i think we should make the change as "diff" can be executed before "apply". for "node" we can do it too, but the upgrade sequence is always to run "apply" first and only then "node". it can be done just in case for consistency.

Reply:
thinking more about this, we should mutate the in cluster config only during "apply" and just mutate the local fetched struct copy during "diff" and "node".

Reply:
updated.